### ISSUE TITLE
"Snooze" (reply-later) feature for any message: Hide this email for x hours/days/weeks/months

### ISSUE THREAD

**Issue by marclaporte** [cypht-org/cypht#572]
Like many (most?) people, my email inboxes and other email folders often contain todos and reminders. When I am completely finished with an email, I will delete it (If I am sure I will never need it again) or I archive it (out of my view but still findable): https://github.com/jasonmunro/cypht-website/issues/24

Cypht nicely permits to [combine views](https://github.com/jasonmunro/cypht-website/issues/25), so this accentuates the challenge of quite a few emails with diverging levels of urgency.

While [inbox zero](https://search.brave.com/search?q=inbox+zero) is elusive, I regularly review emails from most recent to oldest. This "[last in, first out aka LIFO method](https://www.themuse.com/advice/4-strategies-for-keeping-your-inbox-empty-which-one-is-right-for-you)" is debatable logic as older emails can be more important, and they have been waiting longer. However, newer emails sometimes make older emails no longer important, so it makes sense to start with latest information. And [threading will help](https://github.com/jasonmunro/cypht/issues/10). And I refuse to [declare email bankruptcy](https://en.wikipedia.org/wiki/Email_bankruptcy).

[Better filters being a way to deal with information overload](https://www.youtube.com/watch?v=LabqeJEOQyI), let's not analyze the same email again and again (without anything immediately actionable). Examples:
- I am in CC and someone else is supposed to answer. I want to make sure they don't forget because this is important.
- Someone replied to me with a quick confirmation of receipt, and it will take a few days/weeks/months before I can expect a follow-up on this topic
- An important but not urgent email, and I want to make sure it's not lost
- A confirmation of an appointment/online event far away in the future. Ex.: with information on how to access the webinar
- etc.

In cases like this, I would like to snooze the email with choices like 1 day, 1 week, 1 month, pick a date. The snoozed emails should all be findable in a "snoozed" folder (or similar name), with a column for the snoozed-until date. Here, you can un-snooze them (it becomes a normal email), or change the snooze date. At the end of the snooze period, the email should appear in a dedicated section above all other emails. Then, you deal with them, or snooze them again.

It should be possible to "snooze" emails in any folder (sent, draft, etc.). If you send an important email, you can then add yourself a reminder to check up on it in one month (for example).

Why not use the flag tag? The flag tag is indeed useful, but then, you can end up evaluating an email again and again. We really need to hide the email until a specific date for that email.

Alternate names for this feature:
- Snooze
- Defer
- Postpone
- Delay
- Reminder
- ?

Related implementations / feature requests:
- https://github.com/toptensoftware/roundcube-snooze
- https://github.com/nextcloud/mail/issues/2048 which referenced https://github.com/kaosfere/snoozebox
- https://www.getmailbird.com/snooze-your-way-to-inbox-zero/
- https://www.zoho.com/mail/help/email-snooze.html
- https://support.google.com/mail/answer/7622010
- https://www.howtogeek.com/664915/how-to-use-outlook-onlines-snooze-tool/
- https://www.hey.com/features/reply-later/

This feature request is one of many enhancements on the way via the [collaboration with the Tiki project](https://dev.tiki.org/Email-as-a-first-class-citizen)

I look forward to your thoughts to improve the plan before we start coding this.

Thanks!

**Comment by dumblob** [comment]
So basically a **transactional** e-mail **re**delivery after a chosen period.

Yep, I also find this feature useful (though in my case I immediately put it to calendar manually and keep 99% of all emails I receive).

**Comment by nekohayo** [comment]
In case this is relevant: https://datatracker.ietf.org/doc/html/draft-ietf-extra-sieve-snooze-00

**Comment by marclaporte** [comment]
This is a popular feature request: https://github.com/roundcube/roundcubemail/issues/6603

**Comment by marclaporte** [comment]
To be part of the JMAP spec: https://github.com/jmapio/jmap/issues/301

**Comment by marclaporte** [comment]
Progress: https://datatracker.ietf.org/doc/draft-ietf-extra-sieve-snooze/
